src/main/java/Deadline.java

```diff
@@ -0,0 +1,130 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+
+/**
+ * Represents a deadline, which is a type of task.
+ * An deadline has a description and is due by a specific date/time.
+ */
+public class Deadline extends Task {
+    protected String by;
+    protected Date byDate;
+    protected String day;
+    protected String month;
+    protected String year;
+    protected String hour;
+    protected String minute;
+
+    /**
+     * Creates a deadline object.
+     * Converts the String argument for the deadline date/time into a date object.
+     * If the deadline date/time argument is not in the required format, an exception is thrown.
+     *
+     * @param description The string description of the deadline created.
+     * @param by The String of the deadline time/date.
+     * @throws InvalidTaskArgumentDukeException if "by" contains invalid information.
+     */
+    public Deadline(String description, String by) throws InvalidTaskArgumentDukeException {
+        super(description);
+        this.by = by;
+
+        String[] dateTime = by.split(" ");
+        String[] date = dateTime[0].split("/");
+
+        SimpleDateFormat format = new SimpleDateFormat("dd/MM/yyyy hh:mm");
+        try {
+            this.day = date[0];
+            this.month = date[1];
+            this.year = date[2];
+            this.hour = dateTime[1].substring(0, 2);
+            this.minute = dateTime[1].substring(2);
+
+            this.byDate = format.parse(day + "/" + month + "/" + year + " " + hour + ":" + minute);
+        } catch (Exception e) {
+            throw new InvalidTaskArgumentDukeException("☹ OOPS!!! The format of deadline timing is invalid.");
+        }
+
+    }
+
+    /**
+     * Converts the deadline's date object into the required format.
+     *
+     * @return The String of the deadline's date/time in the required format.
+     */
+    public String dateToString() {
+        SimpleDateFormat format;
+
+        if (this.day.equals("1")) {
+            format = new SimpleDateFormat("d" + "'st of '" + "MMMM yyyy" + "', '" + "hh:mma");
+        } else if (this.day.equals("2")) {
+            format = new SimpleDateFormat("d" + "'nd of '" + "MMMM yyyy" + "', '" + "hh:mma");
+        } else if (this.day.equals("3")) {
+            format = new SimpleDateFormat("d" + "'rd of '" + "MMMM yyyy" + "', '" + "hh:mma");
+        } else {
+            format = new SimpleDateFormat("d" + "'th of '" + "MMMM yyyy" + "', '" + "hh:mma");
+        }
+
+        return format.format(this.byDate);
+    }
+
+    public Date getByDate() {
+        return byDate;
+    }
+
+    public void setByDate(Date byDate) {
+        this.byDate = byDate;
+    }
+
+    public String getDay() {
```

Comment:
Hi,

I see that you are trying to make getter and setter for the Deadline object. May i know the use of this setters and getters, as i could not really find the use of these getters and setters in your code. Maybe, i can learn from your point of view. However, just a small input, if you do not really need these methods, you can either not write them or you can have 1 method that will return the date, unless there is really a need to getHour or getMinute, i think defining these methods will make ur cleaned and organized code become a little bit messy.

But overall is good, im just curious about this approach, maybe i can learn from you as well. Cheers!!

Reply:
Thanks for the useful feedback overall, I really appreciate it! 

Regarding the unused getter and setter methods, I realized it may be a mistake on part as I implemented these methods by convention regardless of any use of them. I have learned that it would be bad practice to do so as it may sometimes cause the abstraction barrier to be broken. 

Also, good eye catching the repeated use of the `isExit` method in my `Command` classes. What you suggested made a lot of sense and was something that slipped my mind while coding the `Command` classes. 

Once again, thanks for the constructive comments, they definitely improved my current code!